### DISCUSSION TITLE
The search results differ significantly between the GPU_CAGRA and AUTOINDEX indexes

### DISCUSSION THREAD

**Discussion by youqugit** [milvus-io/milvus#34006] [Q&A and General discussion]
I built GPU_CAGRA and AUTOINDEX indexes for the same data(two same collections). When I perform searches using the same text (with cosine as the evaluation metric), their results differ significantly. For the GPU_CAGRA index, I first normalize the vectors and then calculate the inner product (IP).Below are the parameters I used for building the indexes and performing the searches. 
```
{
"index_type": "AUTOINDEX",
"metric_type": "COSINE",
"params": {},
}

{
      "metric_type": "IP",
      "index_type": "GPU_CAGRA",
      "params": {
          'intermediate_graph_degree': 128,
          'graph_degree': 64
      }
}

# search_params
{"metric_type": "COSINE", "params": {"ef": 100}}

{
    "metric_type": "IP",
    "params": {
        "itopk_size": 128,
        "search_width": 4,
        "min_iterations": 0,
        "max_iterations": 0,
        "team_size": 0
    }
}

```
When I use GPU_CAGRA, only the first or second result is semantically similar to the original text, while the other results differ significantly in meaning. However, all results from AUTOINDEX are more reasonable.
Is this phenomenon normal?

**Comment by youqugit** [comment]
GPU_CAGRA uses the milvus [v2.4.4-gpu] version
AUTOINDEX uses the milvus [v2.4.0-rc.1] version
When searching with a single vector, they take the same amount of time.
Or is there a problem with my vector normalization calculations?

**Comment by yhmo** [comment]
How many rows in the collection?
What is the dimension value?

**Reply by youqugit** [reply]
about 3000 rows
1024 dimension

**Comment by presburger** [comment]
@youqugit hi，Currently, in the version of RAFT integrated with Milvus, there is a bug when using IP distance with the CAGRA index. If the vectors are already normalized, you can temporarily use L2 distance instead, as it is mathematically equivalent.
